wireapp/wire-ios-request-strategy
# fix: unable to remove a service from a conversation FS-637

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-637" title="FS-637" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />FS-637</a>  [iOS] Unable to remove a service
  </td></table>
  <br />
 


----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Regression tests show that since API version was merged, we are unable to remove a service (aka bot) form a conversation.

### Causes

When receiving the response payload from the request to remove the service from the conversation, a decoding error occurred and so the service could not be removed.

For some reason there is a different decoding path is the participant being removed is a service user. There is also a different endpoint to call when removing a service (as opposed to a normal user): `DELETE /conversations/<conversationID>/bots/<userID>`. This endpoint is apparently no longer supported (or even exists). I couldn't find any more info about this.

### Solutions

First, remove usage of the "bot" endpoint, only use `DELETE /conversations/<conversationID>/members/<userID>`. It still works for bots! Fortunately now there is only a single response model to decode, so remove the bot specific response payload.

### Testing

- Updated unit test
- Did manual tests for api version 0 and 1 to verify both versions work.

#### How to Test

1. Add a bot to a conversation
2. Remove the bot from the conversation.
3. Verify it can be removed.  

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
